src/vim9type.c

```diff
@@ -907,6 +907,8 @@ check_type_maybe(
 		return MAYBE;	// use runtime type check
 	    if (actual->tt_type != VAR_OBJECT)
 		return FAIL;	// don't use tt_class
+	    if (actual == &t_object)
+		return OK;	// NOTE: t_object *is* a null_object
 
```

Comment:
With the support for the null_object, the existing Test_object_not_set() test in test_vim9_class.vim is failing with the `E1360: Using a null object` error instead of the `E1012: Type mismatch; expected object<Background> but got object<Unknown>` error.  You need to update that test with the new error.